docs/public/workbench/codeserver_v1beta1_workspacekind.yaml

```diff
@@ -0,0 +1,365 @@
+    ##  - spec for PodSecurityContext:
+    ##    https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#podsecuritycontext-v1-core
+    ##
+    securityContext:
+      runAsUser: 0 # s6 需要 root，同样下面 containerSecurityContext 需要注释
+      fsGroup: 100
+
+    ## container SecurityContext for Workspace Pods (MUTABLE)
+    ##  - spec for SecurityContext:
+    ##    https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#securitycontext-v1-core
+    ##
+    # containerSecurityContext:
+    #   allowPrivilegeEscalation: false
+    #   capabilities:
+    #     drop:
+    #       - ALL
+    #   runAsNonRoot: true
+
+    ## ==============================================================
+    ## WORKSPACE OPTIONS
+    ##  - options are the user-selectable fields,
+    ##    they determine the PodSpec of the Workspace
+    ## ==============================================================
+    options:
+
+      ##
+      ## About the `values` fields:
+      ##  - the `values` field is a list of options that the user can select
+      ##  - elements of `values` can NOT be removed, only HIDDEN or REDIRECTED
+      ##     - this prevents options being removed that are still in use by existing Workspaces
+      ##     - this limitation may be removed in the future
+      ##  - options may be "hidden" by setting `spawner.hidden` to `true`
+      ##     - hidden options are NOT selectable in the Spawner UI
+      ##     - hidden options are still available to the controller and manually created Workspace resources
+      ##  - options may be "redirected" by setting `redirect.to` to another option:
+      ##     - redirected options are NOT shown in the Spawner UI
+      ##     - redirected options are like an HTTP 302 redirect, the controller will use the target option
+      ##       without actually changing the `spec.podTemplate.options` field of the Workspace
+      ##     - the Spawner UI will warn users about Workspaces with pending restarts
+      ##
+
+      ## ============================================================
+      ## IMAGE CONFIG OPTIONS
+      ##  - SETS: image, imagePullPolicy, ports
+      ## ============================================================
+      imageConfig:
+
+        ## spawner ui configs
+        ##
+        spawner:
+
+          ## the id of the default option
+          ##  - this will be selected by default in the spawner ui
+          ##
+          default: "codeserver_012"
+
+        ## the list of image configs that are available
+        ##
+        values:
+
+          ## ================================
+          ## EXAMPLE 1: code server
+          ## ================================
+          - id: "codeserver_012"
+            spawner:
+              displayName: "base-codeserver:v0.1.2"
+              description: "Custom Code server"
+            spec:
+              image: "build-harbor.alauda.cn:60070/mlops/base-codeserver:v0.1.2"
```

Comment:
Use a image place holder and let cluster admins to edit the image url to the actual value for the local cluster.